5_databases/week3/mealSharing.sql

```diff
@@ -0,0 +1,200 @@
+INSERT INTO Meal (`title`, `description`,`location`, `when`,`max_reservations`, `price`,`created_date`) 
+VALUES ( 'Beans stew and rice' , 'rice , bean stew', 'Ribe','2023-11-28 18:30:00', 7 , 100, '2023-11-28');
+INSERT INTO Meal (`title`, `description`,`location`, `when`,`max_reservations`, `price`,`created_date`) 
+VALUES ( 'Lasagna' , 'soya, cabbages, special sauce, corn', 'Alborg','2023-11-28 13:15:00', 5 , 135, '2023-11-28');
+
+-- Adding to Reservation
+INSERT INTO Reservation (`number_of_guests`, `meal_id`,`created_date`, `contact_phonenumber`,`contact_name`, `contact_email`) 
+VALUES ( 1 , 5 ,'2023-11-28',  null ,'Christine', Null );
+INSERT INTO Reservation (`number_of_guests`, `meal_id`,`created_date`, `contact_phonenumber`,`contact_name`, `contact_email`) 
+VALUES ( 3 , 4,'2023-11-28',  004511223344 ,'Lise', 'Lise@gmail.com' );
+INSERT INTO Reservation (`number_of_guests`, `meal_id`,`created_date`, `contact_phonenumber`,`contact_name`, `contact_email`) 
+VALUES ( 2 , 3,'2023-11-28',  004599887766 ,'Yanal', 'Yanal@gmail.com' );
+
+-- Adding to review
+INSERT INTO Review(`title`, `description`,`meal_id`, `stars`,`created_date`) 
+VALUES ('Not Good' ,'It was salty',  5 ,2, Null );
+INSERT INTO Review(`title`, `description`,`meal_id`, `stars`,`created_date`)
+VALUES ( 'Good' ,NULL ,  7 ,5, '2023-11-28' );
+INSERT INTO Review (`title`, `description`,`meal_id`, `stars`,`created_date`)
+VALUES ( 'Very good' ,'very tasty ',  4 ,5, '2023-11-28' );
+
+-- Get meals that have a price smaller than a specific price fx 90
+SELECT * 
+FROM Meal
+WHERE price < 90;
+
+-- Get meals that still have available reservations
+SELECT * 
+FROM Meal
+WHERE max_Reservations > (SELECT SUM(number_of_guests)
+                          FROM Reservation
+                          WHERE Meal_id = Meal.id);
+            
+-- Get meals that partially match a title. 'Rød grød med' will match the meal with the title 'Rød grød med fløde'
+SELECT * 
+FROM Meal
+WHERE title LIKE '%Rød grød med%';
+
+-- Get meals that have been created between two dates
+SELECT * 
+FROM Meal
+WHERE created_date 
+BETWEEN '2023-11-25' AND '2023-12-16';
+
+-- Get only a specific number of meals fx return only 3 meals
+SELECT * 
+FROM Meal
+LIMIT 3;
+
+-- Get the meals that have good reviews
+SELECT  Meal.id, Meal.title, Review.title, Review.stars 
+FROM Meal, Review
+WHERE Review.stars > 4 AND Meal.id = Review.Meal_id;
+
+-- Get reservations for a specific meal sorted by created_date
+SELECT Meal.title, Reservation.Meal_id, SUM(Reservation.Number_of_guests), Reservation.Created_date AS Res_created_date
+FROM Reservation
+JOIN Meal ON (Meal.id=Reservation.Meal_id) WHERE 1
+GROUP BY Meal.title
+ORDER BY Reservation.created_date;
+
+-- Sort all meals by the average number of stars in the reviews
+SELECT Meal.*, AVG(Review.stars) AS avg_stars 
+FROM Meal
+JOIN Review ON Meal.id= Review.Meal_id
+GROUP BY Review.Meal_id
+ORDER BY avg_stars;
```

Comment:
Your user calls you and says that not all meals show up in the output. Especially newer meals seems to be missing 😮 What could be the issue and how do we fix it?